### ISSUE TITLE
[BUG] TARDIS interaction with unloaded console leads to client crash

### ISSUE THREAD

**Issue by einsteinK** [DarkholmeTenk/TardisModIssues#489]
Rightclicking TARDIS blocks may lead to a crash with this crash report:

``` text
Time: 28/03/16 22:04
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
    at tardis.common.blocks.ColorableBlock.getIterator(ColorableBlock.java:61)
    at tardis.common.blocks.ColorableBlock.getColoringIterator(ColorableBlock.java:71)
    at io.darkcraft.darkcore.mod.abstracts.AbstractBlock.func_149727_a(AbstractBlock.java:469)
    at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
    at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1447)
    at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1953)
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
    at tardis.common.blocks.ColorableBlock.getIterator(ColorableBlock.java:61)
    at tardis.common.blocks.ColorableBlock.getColoringIterator(ColorableBlock.java:71)
    at io.darkcraft.darkcore.mod.abstracts.AbstractBlock.func_149727_a(AbstractBlock.java:469)
    at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
    at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1447)

-- Affected level --
Details:
    Level name: MpServer
    All players: 1 total; [GCEntityClientPlayerMP['SpaceTimeVortex'/609526, l='MpServer', x=1,43, y=69,62, z=-67,16]]
    Chunk stats: MultiplayerChunkCache: 81, 81
    Level seed: 0
    Level generator: ID 00 - default, ver 1. Features enabled: false
    Level generator options: 
    Level spawn location: World: (9,70,0), Chunk: (at 9,4,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
    Level time: 248402264 game time, 6001 day time
    Level dimension: 0
    Level storage version: 0x00000 - Unknown?
    Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
    Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
    Forced entities: 1 total; [GCEntityClientPlayerMP['SpaceTimeVortex'/609526, l='MpServer', x=1,43, y=69,62, z=-67,16]]
    Retry entities: 0 total; []
    Server brand: BungeeCord (git:BungeeCord-Bootstrap:1.8-SNAPSHOT:3f991ad:unknown) <- kcauldron,cauldron,craftbukkit,mcpc,fml,forge
    Server type: Non-integrated multiplayer server
Stacktrace:
    at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:373)
    at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2444)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:927)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

After some decompiling fun, I discovered the cause.

``` java
      // tardis.common.blocks.ColorableBlock.getIterator, line 60-61
      CoreTileEntity cte = Helper.getTardisCore(pos.world);
      SchemaCoreTileEntity scte = cte.getSchemaCore(pos);
```

getTardisCore returns the tile entity _(right? I don't know, you do)_ of the Time Rotor thingy above the console, at position (0,70,0).
Apparently, when the console chunk is unloaded, that'll return null, leading to this happening.
A `cte != null` check might fix the crash, although doing TARDIS stuff isn't possible.
In that case, you can at least display a message "The console is unloaded" or something.

I'm pretty sure that the TARDIS console room is auto-chunkloaded, especially due to this.
I even put the Chunk Loader Transthingies on a roundel in every chunk my console is part of.
Apparently, the server I'm on doesn't like TARDIS chunk loading, which I discovered.
Some clues:
- When standing next to the TARDIS Exterior, recalling initiates, but doesn't finish.
  The TARDIS fades out, but never lands, probably because the console unloaded.
  When I teleport to the TARDIS (RFTools teleporter), it continues its journey and lands.
- I went and stood as far from the console as possible, while still seeing it.
  When I rightclicked a block, nothing happened, as is expected.
  I then walked a bit further away from the console, until it disappeared from my view.
  I rightclicked the **same** block, which then crashed me.

Is there any way to make the TARDIS mod's chunkloading be unoverrideable by other stuff?
Another solution would be not using a block, since as shown above, it can actually unload.

I also just realised: I play on a server, but the crash happens clientside.
That means my client can't get the TardisCore at (0,70,0) right?
Is it that the block isn't clientside chunkloaded, ignoring whether it is on the server?
I just tested it: It also crashes on the client if the console is outside the render distance.
This doesn't happen if I put chunk loaders on the roundels around the console.
_(It still crashes on the server though)_

**Comment by FoxPotato** [comment]
I didn't read it all, because i'm lazy and tired rn. But we've had these kind of problems before. Like the sky turning to day when the console room gets unloaded and stuff. 
and the TARDIS not recalling is probably indeed because it's unloaded, so the controls can't be changed. Which is how it works, the recall just changes the set controls and initializes the take off sequence.

**Comment by DarkholmeTenk** [comment]
Chunkloading doesn't actually do anything clientside if you're playing on a server.
Also, this has been fixed #454 
Use the jenkins tardismod/darkcore both client and server side and it should be fixed.
